kratos/modeler/operation/find_contacts_in_skin_model_part.cpp

```diff
@@ -0,0 +1,250 @@
+            rContactContainer.mConditionsMap[temporary_color].push_back(rCondition.Id());
+            for (const auto& r_node : rCondition.GetGeometry()) {
+                rContactContainer.mNodeMap[temporary_color].push_back(r_node.Id());
+            }
+        }
+    }
+}
+
+void FindContactsInSkinModelPart::GetClosestContactColorFromNeighbours(
+    const std::vector<int>& rContactColors,
+    const std::vector<int>& rNeighbourColors,
+    const std::vector<array_1d<std::size_t, 3>>& rNeighbourIndexes,
+    const array_1d<double,3>& rConditionCenter,
+    const array_1d<double,3>& rConditionNormal,
+    const CartesianMeshColors& rColors,
+    int& rPairingStatus,
+    double& rMinDistance,
+    int& rTemporaryColor) const
+{
+    constexpr double angle_threshold = 0.3;
+    int i = 0;
+    for(auto& ind: rNeighbourIndexes){
+        auto cell_center = rColors.GetCenterOfElement(ind[0], ind[1], ind[2]);
+        const int neigh_color = rNeighbourColors[i];
+        array_1d<double,3> dist = cell_center - rConditionCenter;
+        double dist_modulus = norm_2(dist);
+        array_1d<double,3> unit_dist;
+        if(dist_modulus > 1e-9) {
+            unit_dist = dist/dist_modulus;
+        }
+        const double projection = inner_prod(unit_dist, rConditionNormal);
+        bool color_is_contact = std::find(
+            rContactColors.begin(),
+            rContactColors.end(), neigh_color) != rContactColors.end();
+
+        if(color_is_contact &&
+        projection > angle_threshold &&
+        dist_modulus < rMinDistance) {
+            rPairingStatus = 2;
+            rMinDistance = dist_modulus;
+            rTemporaryColor = neigh_color;
+        }
+        i++;
+    }
+}
+
+void FindContactsInSkinModelPart::GetCellNeihgbourColors(
+    const array_1d<std::size_t,3>& rCellIndexes,
+    std::vector<int>& rNeighbourColors,
+    const CartesianMeshColors& rColors) const
+{
+    rNeighbourColors.clear();
+    int i_tmp,j_tmp,k_tmp;
+    for (int i = -1; i < 2; i++) {
+        i_tmp = rCellIndexes[0]+i;
+        for (int j = -1; j < 2; j++) {
+            j_tmp = rCellIndexes[1]+j;
+            for (int k = -1; k < 2; k++) {
+                k_tmp = rCellIndexes[2]+k;
+                if (!(i == 0 && j == 0 && k == 0) && IsElementInsideBounds(rColors, i_tmp, j_tmp, k_tmp)) {
+                    rNeighbourColors.push_back(std::lround(rColors.GetElementalColor(i_tmp, j_tmp, k_tmp)));
+                }
+            }
+        }
+    }
+}
+
+void FindContactsInSkinModelPart::GetCellNeihgbourColorsConnectedByFace(
+    const array_1d<std::size_t,3>& rCellIndexes,
+    const CartesianMeshColors& rColors,
+    std::vector<int>& rNeighbourColors,
+    std::vector<array_1d<std::size_t, 3>>& rNeighbourIndexes) const
+{
+    rNeighbourColors.clear();
+    rNeighbourIndexes.clear();
+    array_1d<int,6> i{-1,1,0,0,0,0};
+    array_1d<int,6> j{0,0,-1,1,0,0};
+    array_1d<int,6> k{0,0,0,0,-1,1};
+    for (int index = 0; index < 6; index++) {
```

Comment:
`index` appears marked in the highlighter as a reserved word/function. I am not aware of if its an intrinsic or smth, but maybe its a good idea to change.